controller/src/test/java/htw/kbe/maumau/controller/service/ViewServiceImplTest.java

```diff
@@ -3,144 +3,145 @@
+    @DisplayName("should return given number from user input and only accept numbers greater than or equal to minimum " +
+            "or less than or equal to maximum")
+    public void getNumberOfPlayer() {
+        String[] userInput = new String[]{
+                "invalid_string",
+                "-",
+                "0",
+                "5",
+                "1",
+                "4"
+        };
+        int expected = 4;
+        System.setIn(new ByteArrayInputStream(String.join("\n", userInput).getBytes()));
+
+        assertEquals(expected, service.getNumberOfPlayer());
     }
 
     @Test
-    @DisplayName("should create a new list of given player names")
+    @DisplayName("should create a new list of given player names and only accepts names greater than 4 and" +
+            "or less than 15 characters")
     public void getPlayerNames() {
-        List<String> expectedPlayerNames = List.of("Philipp", "Jasmin", "Richard", "Maria");
-
-        when(mockedUtilitiesHelper.getPlayerName()).thenReturn(expectedPlayerNames.get(0),
-                expectedPlayerNames.get(1),
-                expectedPlayerNames.get(2),
-                expectedPlayerNames.get(3));
+        List<String> expectedPlayerNames = List.of("Philipp");
+        String[] input = new String[]{
+                " ",
+                "5",
+                "No",
+                "-",
+                "RichardJasminAndPhillip",
+                "Philipp",
+                "anything_after_Richard_will_be_ignored"
+        };
+        System.setIn(new ByteArrayInputStream(String.join("\n", input).getBytes()));
+
+        List<String> actualPlayerNames = service.getPlayerNames(1);
 
-        List<String> actualPlayerNames = service.getPlayerNames(4);
-
-        verify(mockedUtilitiesHelper, times(4)).getPlayerName();
-        assertEquals(4, actualPlayerNames.size());
         assertEquals(expectedPlayerNames, actualPlayerNames);
     }
 
     @Test
     @DisplayName("should create a Map with chosen Card to play and chosen 'mau' state")
     public void getChosenCardAndMauState() {
-        Player player = new Player("Jasmin");
+/*        Player player = new Player("Jasmin");
         player.setHandCards(List.of(new Card(Suit.CLUBS, Label.JACK), new Card(Suit.SPADES, Label.ASS)));
-        when(mockedUtilitiesHelper.getChosenNumber(anyInt(), anyInt())).thenReturn(2, 1);
+        String[] input = new String[]{
+                "2",
+                "1"
+        };
+        InputStream in = new ByteArrayInputStream(String.join("\n", input).getBytes());
+        System.setIn(in);
 
         Map<Card, Boolean> actualMap = service.getPlayedCard(player);
 
         assertTrue(actualMap.containsKey(new Card(Suit.SPADES, Label.ASS)));
-        assertEquals(true, actualMap.get(new Card(Suit.SPADES, Label.ASS)));
-        verify(mockedUtilitiesHelper, times(2)).getChosenNumber(anyInt(), anyInt());
+        assertEquals(true, actualMap.get(new Card(Suit.SPADES, Label.ASS)));*/
```

Comment:
@RiFrost Hast du eine Idee wie wir das testen können? Leider wird bei erneutem Methodenaufruf (innerhalb von getPlayCard()) der Input wieder gelöscht und wir bekommen eine Exception (Nullpointer oder NotSuchELement). Als die Methoden noch public waren und in der anderen Klasse, konnte man das natürlich deutlich einfacher testen, aber wir sollen das ja ändern...